dashboard/app/models/user.rb

```diff
@@ -605,10 +605,36 @@ def user_level_locked?(script_level, level)
     user_level.nil? || user_level.locked?(script_level.stage)
   end
 
+  # Returns the next script_level for the next progression level in the given
+  # script that hasn't yet been passed, starting its search at the last level we submitted
   def next_unpassed_progression_level(script)
     user_levels_by_level = user_levels_by_level(script)
 
-    script.script_levels.detect do |script_level|
+    # Find the user level that we've most recently had progress on
+    user_level = user_levels_by_level.values.flatten.sort_by!(&:updated_at).last
+
+    script_level_index = 0
+    if user_level
+      last_script_level = user_level.level.script_levels.where(script_id: script.id).first
+      script_level_index = script.script_levels.find_index(last_script_level) if last_script_level
```

Comment:
You can use `last_script_level.chapter - 1` instead of doing a `find_index` here.